tests/testsuite/build_script.rs

```diff
@@ -3925,14 +3929,16 @@ fn warnings_emitted_when_build_script_panics() {
 }
 
 #[cargo_test]
-fn warnings_hidden_for_upstream() {
+fn errors_and_warnings_hidden_for_upstream() {
```

Comment:
The name of this test implies errors are hidden but they aren't.

In general, I'm feeling we should probably test errors and warnings separately.

Reply:
Ah, this looks to be part of a larger misunderstanding, see https://github.com/rust-lang/cargo/pull/14435#discussion_r1725100785